core/modules/layout/layout.module

```diff
@@ -2741,3 +2746,126 @@ function _layout_clean_custom_css($classes = array()) {
 function layout_field_delete_instance($instance) {
   cache('layout_path')->flush();
 }
+
+/**
+ * Implements hook_layout_load_by_router_item_alter().
+ */
+function layout_layout_load_by_router_item_alter(&$layouts, $router_item) {
```

Comment:
It's probably OK, but do we need to use the hook inside Layout module? Or is this to not interfere with the original code in layout_load_multiple_by_router_item()?

Reply:
It could maybe be moved to other functions such as `layout_get_path_layout_names()` and so on. Just haven't figured out how to split it.

Reply:
Agreed this alter shouldn't be done. `layout_get_path_layout_names()` seems right.

Reply:
Easier said than done.

Reply:
I don't think it's safe to put it in `layout_get_path_layout_names()` which is also called by `LayoutMenuItem::reassign()`. This method "This checks all of the layouts that match this menu item's path, and renames this menu item to match the highest weighted layout (which is usually the fallback layout at a particular path)." I don't think we want to put additional paths in this list because they shouldn't be associated with the menu item.